ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -1022,4 +1046,53 @@ public static String stripSlashAndExtensionFromId(String blobIdWithExtension) {
     int endIndex = extensionIndex != -1 ? extensionIndex : blobIdWithExtension.length();
     return blobIdWithExtension.substring(startIndex, endIndex);
   }
+
+  /**
+   * Class to encode decode kv header values.
+   */
+  public static class KVHeaderValueEncoderDecoder {
+    private static String DELIM = "; ";
+    private static String KV_SEPERATOR = "=";
+
+    /**
+     * Encode a map of {@link String}s as http header value.
+     * @param map {@link Map} of {@link String}s to encode.
+     * @return encoded http header value.
+     */
+    public static String encodeKVHeaderValue(Map<String, String> map) {
+      Objects.requireNonNull(map);
+      String value = "";
+      for (Map.Entry<String, String> entry : map.entrySet()) {
+        value += entry.getKey() + KV_SEPERATOR + entry.getValue() + DELIM;
+      }
```

Comment:
we could just use join method from collectors to do this
```
String  value = map.entrySet().stream().map(e -> e.getKey() + KV_SEPERATOR + e.getValue()).collect(Collectors.joining(DELIM));
```

Reply:
+1